### PR TITLE
Split up make file targets for easier testing

### PR DESCRIPTION
**Pull request by rshriram**
This PR splits up targets in the Makefile (esp the CLI and sidecar) such that there are explicit makefile targets to build the components on respective platforms.

This simplifies testing and also eliminates one unnecessary build.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -125,14 +127,36 @@ build.k8srules:
...
+build.cli.windows: tools.go-bindata
+	@echo "--> building cli for Windows"
 	@go-bindata -pkg=utils -prefix "./cli" -o ./cli/utils/i18n_resources.go ./cli/locales
-	@$(foreach GOOS, $(TARGET_OS), env GOOS=$(GOOS) GOARCH=amd64 go build -ldflags '$(subst -linkmode external,,$(LDFLAGS))' -o $(BINDIR)/$(CLI_APP_NAME)-$(GOOS) ./cmd/cli/;) # Remove "-linkmode external" flag during build
-	@go build $(BUILDFLAGS) -ldflags '$(subst -linkmode external,,$(LDFLAGS))' -o $(BINDIR)/$(CLI_APP_NAME) ./cmd/cli/ # build an additional binary for the current OS
-	@mv $(BINDIR)/$(CLI_APP_NAME)-windows $(BINDIR)/$(CLI_APP_NAME)-windows.exe # add extension to windows binary
+    @$(env GOOS=windows GOARCH=amd64 go build $(BUILDFLAGS) -o $(BINDIR)/$(CLI_APP_NAME)-windows.exe ./cmd/cli/;)
 	@goimports -w ./cli/utils/i18n_resources.go
 
+# build.cli: tools.go-bindata
```
Comment (gicaceres) [inline Makefile:148]: Delete this block.

File: Makefile [hunk 2]
```diff
@@ -125,14 +127,36 @@ build.k8srules:
...
+build.cli.windows: tools.go-bindata
+	@echo "--> building cli for Windows"
 	@go-bindata -pkg=utils -prefix "./cli" -o ./cli/utils/i18n_resources.go ./cli/locales
-	@$(foreach GOOS, $(TARGET_OS), env GOOS=$(GOOS) GOARCH=amd64 go build -ldflags '$(subst -linkmode external,,$(LDFLAGS))' -o $(BINDIR)/$(CLI_APP_NAME)-$(GOOS) ./cmd/cli/;) # Remove "-linkmode external" flag during build
-	@go build $(BUILDFLAGS) -ldflags '$(subst -linkmode external,,$(LDFLAGS))' -o $(BINDIR)/$(CLI_APP_NAME) ./cmd/cli/ # build an additional binary for the current OS
-	@mv $(BINDIR)/$(CLI_APP_NAME)-windows $(BINDIR)/$(CLI_APP_NAME)-windows.exe # add extension to windows binary
+	@GOOS=windows GOARCH=amd64 go build $(BUILDFLAGS) -o $(BINDIR)/$(CLI_APP_NAME)-windows.exe ./cmd/cli/
 	@goimports -w ./cli/utils/i18n_resources.go
 
+# build.cli: tools.go-bindata
```
Comment (ijsnellf) [inline Makefile:148]: Delete this instead of leaving it commented out?

**Review by ijsnellf — APPROVED** [review]
LGTM, but @zcahana should probably review as well.

**Review by gicaceres — APPROVED** [review]
LGTM

File: Makefile (hunk 2 above)
Comment (rshriram) [reply Makefile:148]: I kept this intentionally. The Makefile can use some more work in future, to coalesce targets. Didn't have the patience to do it all in one go. The commented code snippet would serve as a useful reference when we want to collapse the build targets in future.

**Pull request merged**